src/fs/inode_info.hh

```diff
@@ -85,11 +85,11 @@ struct inode_info {
         }
 
         // Deletes data vectors that are subset of @p data_range and cuts overlapping data vectors to make them
-        // not overlap. @p cut_data_vec_processor is called on each inode_data_vec (including parts of overlapped
-        // data vectors) that will be deleted
+        // not overlap. @p data_vec_cutting_callback is called with three parameters: an inode_data_vec that
+        // intersects with @p data_range and two data vector corresponding to its non-intersecting parts
```

Comment:
I'm not sure if this is clear enough. Can someone else check this? @tropuq

Reply:
I think it is missing information that inode_data_vec is called for every inode_data_vec that intersects data_range and explanation what are those `non-intersecting parts`. Probably it would be good to add information that those two non-intersecting parts can be non-empty only for inode_data_vecs on the edges of data_range.

Reply:
The comments on lines 93,103-106 become redundant after including all this info, don't they?

Reply:
I think they are **not** redundant. Especially the second one, as it clearly illustrates the situation and which variable represents what.
I can see that you are trying to minimize the amount of code. Is that right? Please tell me why do you think these comments are redundant, because I don't see it.

Reply:
> I think it is missing information that inode_data_vec is called for every inode_data_vec that intersects data_range

comment on line 93 doesn't speak about calling data_vec_cutting_callback but it does mention that we're working on every inode_data_vec that intersects data_range
> and explanation what are those `non-intersecting parts`. 

as you have noted, the second comment illustrates those 'non-intersecting parts'
> Probably it would be good to add information that those two non-intersecting parts can be non-empty only for inode_data_vecs on the edges of data_range.

this would be admittedly a new information 
>I can see that you are trying to minimize the amount of code. Is that right? Please tell me why do you think these comments are redundant, because I don't see it.

I'm actually trying to minimize the amount of remaining review, I've asked for clarification because I've already got reviews about comments being redundant and becoming outdated after implementing changes

Reply:
Users of that function probably wouldn't like to read implementation of that function to know how it works so I think all necessary information should be outside the implementation.

Reply:
I'm not saying the comment outside shouldn't be expanded. I'm asking if the comments inside should be removed while I'm expanding it

Reply:
I think it depends on how you describe it outside the function. I would probably remove the comment from the inside if it was almost a copy-paste from the outside.

Reply:
> > I can see that you are trying to minimize the amount of code. Is that right? Please tell me why do you think these comments are redundant, because I don't see it.
> 
> I'm actually trying to minimize the amount of remaining review, I've asked for clarification because I've already got reviews about comments being redundant and becoming outdated after implementing changes

I thought that we want to optimize the output (high quality code that is ready to be merged) not the process to get to the output. I don't think that minimizing number of iterations of the review is the best choice. As the review is very fast now.
Regarding the comments, the comment above function declaration is to help the user of this function when he needs to use it. The comments inside implementation are to help the potential reader understand what is happening inside. One of aspect of why they are useful is their placement -- they are situated alongside code they refer to. And last but not least, comments in the implementation are used to minimize the time that reader needs to understand what is happening on such level that they know or are able to do what they wanted in the first place -- this can be e.g. editing this function, navigating through it to find relevant part (this is very important as it happens a lot, and in this process you don't read the whole implementation, you want as quickly as possible find what you want) or get a comprehensive understanding of what function does and how it does it.
Looking from that perspective, it make sense to leave these comments were they are because they help to achieve above goals. Maybe you don't need them, that is fine. But other people may find them very helpful, and I think it is very probable. As you don't write this code for yourself only, I suggest taking others perspective seriously into consideration, as people are very different and not everyone will understand what is going on there as quickly as you would.

Reply:
@tropuq 
> I think it depends on how you describe it outside the function. I would probably remove the comment from the inside if it was almost a copy-paste from the outside.

That's what I was fearing would happen, but as we can see, there are uses for the inside comments even if they are quite similar

@varqox 
> I thought that we want to optimize the output (high quality code that is ready to be merged)

I've never disagreed with it and I don't know why does it need to be mentioned
> not the process to get to the output. 

As with the amount of code, minimizing the number of review iterations is hardly optimizing
> I don't think that minimizing number of iterations of the review is the best choice. As the review is very fast now.

While it's hard to disagree with that, it's also hard to agree that avoiding adding another review iteration has no benefits, given that we don't have unlimited time
> Regarding the comments, the comment above function declaration is to help the user of this function when he needs to use it. The comments inside implementation are to help the potential reader understand what is happening inside. One of aspect of why they are useful is their placement -- they are situated alongside code they refer to. And last but not least, comments in the implementation are used to minimize the time that reader needs to understand what is happening on such level that they know or are able to do what they wanted in the first place -- this can be e.g. editing this function, navigating through it to find relevant part (this is very important as it happens a lot, and in this process you don't read the whole implementation, you want as quickly as possible find what you want) or get a comprehensive understanding of what function does and how it does it.

Thank you for this fragment, because I believe it answers my question - why are the inside comments useful despite mostly repeating what's in the comment above the function.
> Looking from that perspective, it make sense to leave these comments were they are because they help to achieve above goals. Maybe you don't need them, that is fine. But other people may find them very helpful, and I think it is very probable. As you don't write this code for yourself only, I suggest taking others perspective seriously into consideration, as people are very different and not everyone will understand what is going on there as quickly as you would.

Isn't taking others perspective into consideration exactly what I'm doing by asking for that perspective instead of going ahead and deleting all the comments that I didn't find necessary?

All in all, I'm having an impression that I've been perceived as trying to reject any change or force my view while all I did was ask for a clarification that would help me write the high quality code that would not need further fixes.

Reply:
> > Looking from that perspective, it make sense to leave these comments were they are because they help to achieve above goals. Maybe you don't need them, that is fine. But other people may find them very helpful, and I think it is very probable. As you don't write this code for yourself only, I suggest taking others perspective seriously into consideration, as people are very different and not everyone will understand what is going on there as quickly as you would.
> 
> Isn't taking others perspective into consideration exactly what I'm doing by asking for that perspective instead of going ahead and deleting all the comments that I didn't find necessary?
> 
> All in all, I'm having an impression that I've been perceived as trying to reject any change or force my view while all I did was ask for a clarification that would help me write the high quality code that would not need further fixes.

Thank you for sharing that. I now see that I misunderstood your intentions and responded inadequately. I feel sad to see you felt sad because of what I wrote, I would have liked to be more aware of your needs.